File: solutions/cpp/raindrops/3/raindrops.cpp
#include "raindrops.h"

namespace raindrops {
    
    std::string convert(int drops) {
    std::string result = "";
    if (drops % 3 == 0) {
        result += "Pling";
    }

    if (drops % 5 == 0) {
        result += "Plang";
    }

    if (drops % 7 == 0) {
        result += "Plong" ;
    }

    if (drops % 3 != 0 && drops % 5 != 0 && drops % 7 != 0) {
        return std::to_string(drops);
    }
    return result;
    }
    
}  // namespace raindrops

File: solutions/fortran/raindrops/1/raindrops.f90
module raindrops
  implicit none
contains

  function convert(i)
    integer :: i
    character(20) :: convert
    character(20) :: temp
    logical :: has_factor
    
    ! Initialize result string
    temp = ''
    has_factor = .false.
    
    ! Check divisibility by 3
    if (mod(i, 3) == 0) then
      temp = trim(temp) // 'Pling'
      has_factor = .true.
    end if
    
    ! Check divisibility by 5
    if (mod(i, 5) == 0) then
      temp = trim(temp) // 'Plang'
      has_factor = .true.
    end if
    
    ! Check divisibility by 7
    if (mod(i, 7) == 0) then
      temp = trim(temp) // 'Plong'
      has_factor = .true.
    end if
    
    ! If not divisible by 3, 5, or 7, return the number as string
    if (.not. has_factor) then
      write(temp, '(I0)') i
    end if
    
    convert = temp
    
  end function convert

end module raindrops